### ISSUE TITLE
Articles order using ZDKHelpCenterProvider.searchArticles with ZDKHelpCenterSearch

### ISSUE THREAD

**Issue by Sky** [zendesk/zendesk_sdk_ios#131]
How to control order of articles returned by this request?
The order is different each request. ZDKHelpCenterArticle.position of all items are 0.
I need them be ordered as in Zendesk Help Center.
Help Center order settings for Articles in this Section is set to Manual.

**Comment by pierrezen** [comment]
Hi Sky,
The search provider is returning articles sorted by relevancy, not by position. Are you experiencing some inconsistency in the order of the articles with the same query ?

**Comment by Sky** [comment]
Actually some of the articles can change order with 2 requests each after another.
But why the position property of those articles are 0, I can't reorded them after fetch?
Does Manual articles order in section settings of Help Center control this property and articles order in general? If not then what is the purpose of this settings at all?

**Comment by Sky** [comment]
Actually I checked that your API returns right position information from server but I don't know why you don't set it to ZDKHelpCenterArticle object property that is always 0.
I made custom request with ordering by position which is supported by API.

**Comment by schlan** [comment]
Hi @Sky,
As we are using the HelpCenter Rest API in our SDK, we are inheriting their behaviour. 
According to the documentation, the default value of `position` is `0`. [HelpCenter API - Articles](https://developer.zendesk.com/rest_api/docs/help_center/articles#json-format)

Can you share the request you are using that results into non null `position` information?

**Comment by Sky** [comment]
I think I am also using rest API. There is another one? :)
The default value is 0 but default value is the case when there is no real value :) In my case there is value, because I use Manual articles ordering in Help Center, so I think by design this property is exactly for this purposes.

```
[[ZDKAPIDispatcher instance] executeRequest: ^NSMutableURLRequest *{
            NSString* requestURLString = [NSString stringWithFormat: @"%@/api/v2/help_center/sections/%@/articles.json?sort_by=position&sort_order=asc", kZendeskURLString, section.sid];
            NSMutableURLRequest* request = [[NSMutableURLRequest alloc] initWithURL: [NSURL URLWithString: requestURLString]];
            return request;
        } onSuccess: ^(ZDKDispatcherResponse* result) {
            NSError* error = nil;
            NSArray* items = nil;
            if (result.data.length) {
                NSDictionary* JSONObject = [NSJSONSerialization JSONObjectWithData: result.data options: 0 error: &error];

                if ([JSONObject isKindOfClass: NSDictionary.class] && JSONObject.count) {
                    ZDKHelpCenterParser* parser = [ZDKHelpCenterParser new];
                    items = [parser parseArticles: JSONObject];
                }
            }
            
            // cache
            if (items.count) {
                [AFHelpCache setCache: items forKey: section.sid class: ZDKHelpCenterArticle.class];
            }
            
            items = [self filterArticlesByLabels: items];
            
            completion(items, error);
        } onError: ^(NSError *error) {
            completion(nil, error);
        } requiresAuth: NO];
```

**Comment by baz8080** [comment]
Hi Sky,

TL;DR - position makes sense for _listing_, not for _searching_

Your example above is for listing, but `ZDKHelpCenterProvider. searchArticles` uses `/api/v2/help_center/articles/search.json`.

Searching is an inherently dynamic and nondeterministic operation. `position` is 0 in a search because it doesn't make sense to have it any other way. The only way that search would ever rely on `position` is if its algorithm used it to as a factor in determining relevancy.

Thanks,
Barry.

**Comment by Sky** [comment]
It's make sense, but not in my case. I use search to filter articles in section by labels (actually this is one of cases that described in labels description in Help Center) because there is no other way to do that using SDK (filter articles by labels in section), so I still need right order and position property.

I think it will be a good addition to add "sort by" and "sort order" to ZDKHelpCenterSearch.

**Comment by baz8080** [comment]
Hi Sky,

Thanks for the clarification. Could you please detail _exactly_ what you would like in our APIs? For example are you looking to list all article in a given section with specific labels?

This makes it easy for us to present it to our PM to get it properly identified in the backlog.

Thanks,
Barry.

**Comment by Sky** [comment]
"are you looking to list all article in a given section with specific labels?"
This is exactly I described in my comment. But I need to persist the order from Help Center.

**Comment by baz8080** [comment]
@Sky I'm going to close this issue as we don't have traction on it. Please feel free to reopen it with some examples and we can try to help.
